app/scripts/metamask-controller.js

```diff
@@ -595,15 +601,19 @@ export default class MetamaskController extends EventEmitter {
 
     let additionalKeyrings = [];
     if (!isManifestV3) {
-      additionalKeyrings = [
+      const additionalKeyringTypes = [
         TrezorKeyring,
         LedgerBridgeKeyring,
         LatticeKeyring,
         QRHardwareKeyring,
       ];
+      additionalKeyrings = additionalKeyringTypes.map((keyringType) =>
+        keyringBuilderFactory(keyringType),
+      );
```

Comment:
The `KeyringController` constructor now takes keyring builder functions rather than classes.

`keyringBuilderFactory` takes the class types and makes them into these builder functions.